src/main/java/com/microsoft/alm/gitcredentialmanager/InsecureStore.java

```diff
@@ -330,7 +349,51 @@ public synchronized void writeCredential(final String targetName, final Credenti
     @Override
     public synchronized void writeToken(final String targetName, final Token token)
     {
+        ensureEnabled();
+
         Tokens.put(targetName, token);
         save();
     }
+
+    /**
+     * Exports all tokens and credentials to the specified {@link ISecureStore}
+     * and then the polite thing to do is to rename the backing file
+     * and prevent further use via this instance.
+     * This will still make a downgrade possible
+     * and the archival of the old file will be the user's responsibility.
+     *
+     * @param destination the {@link ISecureStore} that will replace this {@link InsecureStore}
+     */
+    public synchronized void migrateAndDisable(final ISecureStore destination)
+    {
+        ensureEnabled();
+
+        for (final Map.Entry<String, Token> pair : Tokens.entrySet())
+        {
+            final String targetName = pair.getKey();
+            final Token token = pair.getValue();
+            destination.writeToken(targetName, token);
+        }
+
+        for (final Map.Entry<String, Credential> pair : Credentials.entrySet())
+        {
+            final String targetName = pair.getKey();
+            final Credential credential = pair.getValue();
+            destination.writeCredential(targetName, credential);
+        }
+
+        if (backingFile != null)
+        {
+            final File disabledBackingFile = new File(backingFile.getAbsolutePath() + MIGRATION_SUFFIX);
+            final boolean wasRenamed = backingFile.renameTo(disabledBackingFile);
+            if (!wasRenamed)
+            {
+                final String renameFailureTemplate = "Unable to rename '%1$s' to '%2$s' after migrating its contents.";
+                final String renameFailureMessage = String.format(renameFailureTemplate, backingFile.getAbsolutePath(), disabledBackingFile.getAbsolutePath());
+                throw new Error(renameFailureMessage);
+            }
+        }
+
+        isEnabled = false;
+    }
 }

```

Comment:
> If migration succeeded, is there value in retaining the file?

Because I don't have migration going the other way, keeping the file around makes it easy to downgrade without having to re-authenticate everywhere.  I'll add a TODO about deleting the file once we're satisfied with the stability of the Keychain implementation over the InsecureStore.